storage/flux/reader.go

```diff
@@ -231,276 +230,6 @@ READ:
 	return rs.Err()
 }
 
-type windowAggregateIterator struct {
-	ctx   context.Context
-	s     storage.Store
-	spec  query.ReadWindowAggregateSpec
-	stats cursors.CursorStats
-	cache *tagsCache
-	alloc *memory.Allocator
-}
-
-func (wai *windowAggregateIterator) Statistics() cursors.CursorStats { return wai.stats }
-
-func (wai *windowAggregateIterator) Do(f func(flux.Table) error) error {
-	src := wai.s.GetSource(
-		uint64(wai.spec.OrganizationID),
-		uint64(wai.spec.BucketID),
-	)
-
-	// Setup read request
-	any, err := types.MarshalAny(src)
-	if err != nil {
-		return err
-	}
-
-	var req datatypes.ReadWindowAggregateRequest
-	req.ReadSource = any
-	req.Predicate = wai.spec.Predicate
-	req.Range.Start = int64(wai.spec.Bounds.Start)
-	req.Range.End = int64(wai.spec.Bounds.Stop)
-
-	if wai.spec.WindowEvery != 0 || wai.spec.Offset != 0 {
-		req.Window = &datatypes.Window{
-			Every: &datatypes.Duration{
-				Nsecs: wai.spec.WindowEvery,
-			},
-			Offset: &datatypes.Duration{
-				Nsecs: wai.spec.Offset,
-			},
-		}
-	} else {
-		req.Window = &datatypes.Window{
-			Every: &datatypes.Duration{
-				Nsecs:    wai.spec.Window.Every.Nanoseconds(),
-				Months:   wai.spec.Window.Every.Months(),
-				Negative: wai.spec.Window.Every.IsNegative(),
-			},
-			Offset: &datatypes.Duration{
-				Nsecs:    wai.spec.Window.Offset.Nanoseconds(),
-				Months:   wai.spec.Window.Offset.Months(),
-				Negative: wai.spec.Window.Offset.IsNegative(),
-			},
-		}
-	}
-
-	req.Aggregate = make([]*datatypes.Aggregate, len(wai.spec.Aggregates))
-
-	for i, aggKind := range wai.spec.Aggregates {
-		if agg, err := determineAggregateMethod(string(aggKind)); err != nil {
-			return err
-		} else if agg != datatypes.AggregateTypeNone {
-			req.Aggregate[i] = &datatypes.Aggregate{Type: agg}
-		}
-	}
-
-	aggStore, ok := wai.s.(storage.Store)
-	if !ok {
-		return errors.New("storage does not support window aggregate")
-	}
-	rs, err := aggStore.WindowAggregate(wai.ctx, &req)
-	if err != nil {
-		return err
-	}
-
-	if rs == nil {
-		return nil
-	}
-	return wai.handleRead(f, rs)
-}
-
-const (
-	CountKind = "count"
-	SumKind   = "sum"
-	FirstKind = "first"
-	LastKind  = "last"
-	MinKind   = "min"
-	MaxKind   = "max"
-	MeanKind  = "mean"
-)
-
-// isSelector returns true if given a procedure kind that represents a selector operator.
-func isSelector(kind plan.ProcedureKind) bool {
-	return kind == FirstKind || kind == LastKind || kind == MinKind || kind == MaxKind
-}
-
-func isAggregateCount(kind plan.ProcedureKind) bool {
-	return kind == CountKind
-}
-
-func (wai *windowAggregateIterator) handleRead(f func(flux.Table) error, rs storage.ResultSet) error {
```

Comment:
Why is all this code being deleted? I would not have expected that. Maybe it will become clear as I continue my review.